### ISSUE TITLE
FYI: Ubuntu 18.04 (with python 3.6) and HTTPS: "ENGINE Error in HTTPServer.tick" ... "OSError: [Errno 0] Error"

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#1950]
FYI & FWIW:

Fresh & fully updated Ubuntu 18.04 (with python 3.6) and HTTPS enabled, on starting SABnzbd, logging says: "ENGINE Error in HTTPServer.tick" ... "OSError: [Errno 0] Error"
SABnzbd is still usable via HTTPS, but also with errors in the logging (and thus some in GUI).

Happens with SAB 3.4.0 and 3.2.1, so not something recent.

I don't know if this Ubuntu or Python3.6 or a library (python3.6-ssl or ssl itself). Maybe @jcfp has a thought about it?

Workaround: 

- do not enable HTTPS in SABnzbd. 
- Or ignore the errors ... 
- And/or do not allow HTTPS with older python / ssl version?

Anyway: IMHO a won't fix. Just an issue for reference.

```
>>> ssl.OPENSSL_VERSION
'OpenSSL 1.1.1  11 Sep 2018'
```

**Comment by sanderjo** [comment]
On starting:

```
2021-09-21 13:22:43,746::INFO::[postproc:232] Completed Download Folder /root/Downloads/complete is not on FAT
2021-09-21 13:22:43,747::DEBUG::[__init__:348] Starting downloader
2021-09-21 13:22:43,750::INFO::[notifier:122] Sending notification: Error - [21/Sep/2021:13:22:43] ENGINE Error in HTTPServer.tick
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2021, in start
    self.tick()
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2090, in tick
    s, ssl_env = self.ssl_adapter.wrap(s)
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/ssl_builtin.py", line 67, in wrap
    server_side=True)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
OSError: [Errno 0] Error
 (type=error, job_cat=None)
2021-09-21 13:22:43,750::ERROR::[_cplogging:219] [21/Sep/2021:13:22:43] ENGINE Error in HTTPServer.tick
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2021, in start
    self.tick()
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2090, in tick
    s, ssl_env = self.ssl_adapter.wrap(s)
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/ssl_builtin.py", line 67, in wrap
    server_side=True)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
OSError: [Errno 0] Error

2021-09-21 13:22:43,755::DEBUG::[__init__:351] Starting decoders

````

**Comment by sanderjo** [comment]
On accessing SABnzbd's HTTPS:

```
2021-09-21 13:29:43,449::INFO::[notifier:122] Sending notification: Error - [21/Sep/2021:13:29:43] ENGINE Error in HTTPServer.tick
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2021, in start
    self.tick()
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2090, in tick
    s, ssl_env = self.ssl_adapter.wrap(s)
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/ssl_builtin.py", line 67, in wrap
    server_side=True)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:852)
 (type=error, job_cat=None)
2021-09-21 13:29:43,449::ERROR::[_cplogging:219] [21/Sep/2021:13:29:43] ENGINE Error in HTTPServer.tick
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2021, in start
    self.tick()
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2090, in tick
    s, ssl_env = self.ssl_adapter.wrap(s)
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/ssl_builtin.py", line 67, in wrap
    server_side=True)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:852)

2021-09-21 13:29:43,458::ERROR::[_cplogging:219] [21/Sep/2021:13:29:43] ENGINE Error in HTTPServer.tick
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2021, in start
    self.tick()
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/__init__.py", line 2090, in tick
    s, ssl_env = self.ssl_adapter.wrap(s)
  File "/usr/lib/python3/dist-packages/cherrypy/wsgiserver/ssl_builtin.py", line 67, in wrap
    server_side=True)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: SSLV3_ALERT_CERTIFICATE_UNKNOWN] sslv3 alert certificate unknown (_ssl.c:852)


```

**Comment by jcfp** [comment]
For the `OSError: [Errno 0] Error`, see https://github.com/sabnzbd/sabnzbd/issues/1576#issuecomment-674414212 and the linked upstream python bug. The `sslv3 alert certificate unknown` error only appears on python3.6 or earlier and is an upstream python issue as well, fixed in 3.7.

**Comment by sanderjo** [comment]
I tried the newest of the newest: Ubuntu 12.10 with python 3.9, with these details:

```
Python Version: 	3.9.7 (default, Sep 10 2021, 14:59:43) [GCC 11.2.0] [UTF-8]
OpenSSL: 	OpenSSL 1.1.1l 24 Aug 2021 
```

... and indeed no errors anymore with "Enable HTTPS".

Finally ... after all those years of errors.
